azurerm/internal/services/apimanagement/api_management_resource.go

```diff
@@ -1343,7 +1343,7 @@ func expandApiManagementPolicies(input []interface{}) (*apimanagement.PolicyCont
 	if xmlContent != "" {
 		return &apimanagement.PolicyContract{
 			PolicyContractProperties: &apimanagement.PolicyContractProperties{
-				Format: apimanagement.XML,
+				Format: apimanagement.Rawxml,
```

Comment:
Running the tests for this we get the following diff:

```
  <policies>
    <inbound>
      <set-variable name=\"abc\" value=\"@(context.Request.Headers.GetValueOrDefault(&quot;X-Header-Name&quot;, &quot;&quot;))\" />
      <find-and-replace from=\"xyz\" to=\"abc\" />
    </inbound>
  </policies>
  " => "
  <policies>
    <inbound>
      <set-variable name=\"abc\" value=\"@(context.Request.Headers.GetValueOrDefault(\"X-Header-Name\", \"\"))\" />
      <find-and-replace from=\"xyz\" to=\"abc\" />
    </inbound>
  </policies>
```

To fix that we'll also need to update this line: https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/internal/services/apimanagement/api_management_resource.go#L362

to ignore the differences when comparing the diff - since these get returned encoded from the API, we can do this using this function: https://github.com/terraform-providers/terraform-provider-azurerm/blob/841a9c9f7da454581b6fa761754909e90d3f8b7e/azurerm/internal/services/apimanagement/api_management_product_policy_resource.go#L47

As such could we update this so that this test passes?

Reply:
Thanks for the review and the guidance. I will update the code accordingly.

Reply:
How come the acceptance test didn't fail in GitHub? Are only unit tests performed there?

Reply:
I was unfortunately not able to get the acceptance test to run in my VS Code devcontainer. I think I would have to set a whole bunch of `ARM_*` environment variables somehow.

Reply:
Yeah Github Actions only runs the unit tests - the acceptance tests take longer (and can cost a lot) - which is why we run these manually using another tool